src/core/ext/transport/chttp2/transport/hpack_encoder.cc

```diff
@@ -452,37 +461,46 @@ static void hpack_enc(grpc_exec_ctx *exec_ctx, grpc_chttp2_hpack_compressor *c,
   uint32_t elem_hash;
   size_t decoder_space_usage;
   uint32_t indices_key;
-  int should_add_elem;
+  bool should_add_elem;
+  bool should_add_key;
 
   key_hash = grpc_slice_hash(GRPC_MDKEY(elem));
-  value_hash = grpc_slice_hash(GRPC_MDVALUE(elem));
-  elem_hash = GRPC_MDSTR_KV_HASH(key_hash, value_hash);
 
-  inc_filter(HASH_FRAGMENT_1(elem_hash), &c->filter_elems_sum, c->filter_elems);
+  if (elem_interned) {
+    value_hash = grpc_slice_hash(GRPC_MDVALUE(elem));
```

Comment:
can these be declared here so that they don't escape this scope?

Reply:
Only `value_hash` as the `elem_hash` is used below. Done.